NetworkLayerTests/ListingTests/LoadListingUseCaseTests.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  LoadListingUseCaseTests.swift
+//  NetworkLayerTests
+//
+//  Created by Nicolas Demange on 17/01/2023.
+//
+
+import XCTest
+@testable import NetworkLayer
+
+// MARK: - Class
+
+
+struct ListingItemsMapper {
+    private struct RemoteListingItem: Decodable {
+        let category_id: Int
+        let title: String
+        
+        var listingItem: ListingLoader.Listing {
+            ListingLoader.Listing(category_id: category_id, title: title)
+        }
+    }
+    
+    enum Error: Swift.Error {
+        case invalidData
+    }
+    
+    static func map(_ data: Data, from response: HTTPURLResponse) throws -> [ListingLoader.Listing] {
+        guard response.statusCode == 200, let root = try? JSONDecoder().decode([RemoteListingItem].self, from: data) else {
+            throw Error.invalidData
+        }
+        return root.map { $0.listingItem }
+    }
+}
+
+
+final class ListingLoader {
+    private let client: HTTPClient
+    
+    init(client: HTTPClient) {
+        self.client = client
+    }
+    
+    struct Listing {
+        let category_id: Int
+        let title: String
+    }
```

Comment:
```suggestion
    struct ListingItem {
        let categoryId: Int
        let title: String
    }
```
Tu peux sortir cette struct du Loader.
Il est préférable de nommer cette struct ListingItem, ça permet de mettre en évidence que c'est qu'un seul objet et non un listing.
Attention au "_", en swift le nom d'une variable n'en contient pas.